lib/cypress/population_clone_job.rb

```diff
@@ -100,6 +102,33 @@ def unnumerify(patient)
       end
     end
 
+    def restrict_entry_codes(cloned_patient)
+      # Loop through every data element
+      cloned_patient.qdmPatient.dataElements.each do |entry|
+        # look up if the vendor has a preferred_code_system for the qdmCategory
+        vendor_preference = @test.product.vendor.preferred_code_systems[entry.qdmCategory]
+        # if the vendor does not have a preference, pick the first code
+        if vendor_preference.blank?
+          entry.dataElementCodes = [entry.dataElementCodes[0]]
```

Comment:
Is it possible for an entry to not have a code (for example, coming from QRDA?)-- if so, would it be prudent to check that something exists in [entry.dataElementCodes[0]] before using it here, or should something else in Cypress ensure that a code exists?